File: multi-gammon-core/src/main/java/org/akoshterek/backgammon/move/MoveRecord.java
package org.akoshterek.backgammon.move;

/**
 * @author Alex
 *         date 20.07.2015.
 */
public class MoveRecord {
    /*
     * Common variables
     */
	/* type of the move */
    public MoveType mt = MoveType.MOVE_INVALID;
	/* annotation */
    public String sz;
    /* move record is for player */
    public int fPlayer = 0;
	/* luck analysis (shared between MOVE_SETDICE and MOVE_NORMAL) */
	/* dice rolled */
    public int[] anDice = new int[] {0, 0};
    /* evaluation of the moves */
    public MoveList ml = new MoveList();
    /* "private" data */
    public XMoveGameInfo g = new XMoveGameInfo();	/* game information */
    public XMoveNormal n = new XMoveNormal();		/* chequerplay move */
    public XMoveSetBoard sb = new XMoveSetBoard();	/* setting up board */
}

File: multi-gammon-core/src/main/java/org/akoshterek/backgammon/move/MoveRecord.scala
package org.akoshterek.backgammon.move

/**
  * @author Alex
  *         date 20.07.2015.
  */
class MoveRecord {
  // Common variables
  // type of the move
  var mt: MoveType = MoveType.MOVE_INVALID
  // annotation
  var sz: String = _
  // move record is for player
  var fPlayer: Int = 0

  // luck analysis (shared between MOVE_SETDICE and MOVE_NORMAL)
  // dice rolled
  var _anDice: (Int, Int) = (0, 0)
  def anDice: (Int, Int) = _anDice
  def anDice_=(dice: (Int, Int)): Unit = _anDice = dice.copy()

  // evaluation of the moves
  var ml: MoveList = new MoveList

  // "private" data
  // game information
  var g: XMoveGameInfo = new XMoveGameInfo
  // chequer play move
  var n: XMoveNormal = new XMoveNormal
  // setting up board
  var sb: XMoveSetBoard = new XMoveSetBoard
}